engine/schema/src/com/cloud/dc/dao/DataCenterIpAddressDao.java

```diff
@@ -45,8 +45,11 @@
 
     int countIPs(long dcId, boolean onlyCountAllocated);
 
+    int countIPs(String ipAddress, long podId, long dcId, boolean onlyCountAllocated);
```

Comment:
why two separate coundIPs()s? this can be implemented in one (changed) method can't it? it is not a user facing API or service call

Reply:
The delete operation requires searching for an entry in the op_dc_ip_address_alloc table based on a particular IP address. It checks if that IP is allocated to anyone before deleting. 
It cannot be implemented in one method because the other usages search for a range in a DC or in a POD. Hope this clarifies!!

Reply:
actually it doesn't. It can be one method that acts based upon the presence of values in the parameters. i.e. if there is no ip present it searches for ranges in pods/DCs and if an ip is present (not null) it checks for allocation. If the behaviour is really distinct than so should the name. In that case i suggest you rename it to isAllocatedInPodOrDc(), or maybe countIpAddressUsage() or countIpAddressOccurance?

Reply:
I had exactly the same thought to put everything in one method and pass null values in not-needed parameters. But after looking at the usages of old methods(ranges in pods/DCs), we have to make changes in many places. So, I thought why not use the polymorphism concept to get rid of these many changes. If the community agrees with these many changes, I will go ahead and coalesce them to one method. Else we can always change the name of the current one because now, it has only one usage. Either way, I am good. Please let me know your opinion. Thanks!!

Reply:
@nitin-maharana I'd second the idea to rename the method to `countIpAddressUsage()`

Reply:
let's go with the rename

Reply:
Done!